actors/user/src/main/java/org/sunbird/user/actors/UserProfileReadActor.java

```diff
@@ -884,4 +878,31 @@ private void updateTncInfo(Map<String, Object> result) {
     }
     return new ArrayList<>();
   }
+
+  private void checkUserExistence(Request request) {
+    Map<String,Object>searchMap=new HashMap<>();
+    String value=(String)request.get(JsonKey.VALUE);
+    String encryptedValue = null;
+    try {
+      encryptedValue = this.encryptionService.encryptData(StringUtils.lowerCase(value));
+    } catch (Exception var11) {
+      ProjectCommonException exception = new ProjectCommonException(ResponseCode.userDataEncryptionError.getErrorCode(), ResponseCode.userDataEncryptionError.getErrorMessage(), ResponseCode.SERVER_ERROR.getResponseCode());
+      this.sender().tell(exception, this.self());
+    }
+    searchMap.put((String) request.get(JsonKey.KEY),encryptedValue);
+    ProjectLogger.log("UserProfileReadActor:checkUserExistence: search map prepared "+searchMap,LoggerEnum.INFO.name());
+    CompletableFuture<Response>fut=CompletableFuture.supplyAsync(() -> {
+      Response response=cassandraOperation.getRecordsByProperties(JsonKey.SUNBIRD, JsonKey.USER, searchMap);
+      List<Map<String,Object>>respList=(List)response.get(JsonKey.RESPONSE);
+      Response resp=new Response();
```

Comment:
incase user not found with provided identifier then return 404 error , same as get user by email/phone api

Reply:
@manzarul  in case user not found we are giving 200Ok but `isExists` to false. and isExists to true when user found

Reply:
for not found provide 404

Reply:
done